### PR TITLE
Remove needless sierra saving

### LINKED ISSUE #2355: Refactor snforge to Reuse Existing Sierra File Paths for cairo-profiler Input
### Which component is your feature related to?

Forge

### Feature Request

Context: `cairo-profiler` expects paths to files with serialized `VersionedProgram` (sierra format) in its input file. Right now `snforge` saves relevant sierra programs to separate files when using `--save-trace-data` or `--build-profile` and then puts paths to these files in the input file for `cairo-profiler`. This is a leftover from using `scarb-snforge-test-collector`
 
With Scarb 2.7.0 the sierra files with compiled tests are already present on the disk in the form `cairo-profiler` expects.  we Let's make `snforge` reuse the paths to these files when building input files for `cairo-profiler`.

Relevant code:
- saving sierra program
https://github.com/foundry-rs/starknet-foundry/blob/4a1aea0f9f888c9b4774af897fc010d029d0180c/crates/forge/src/run_tests/test_target.rs#L42

- passing the path to the saved sierra program around:
https://github.com/foundry-rs/starknet-foundry/blob/4a1aea0f9f888c9b4774af897fc010d029d0180c/crates/forge-runner/src/running.rs#L85
to later use it here when generating input file for cairo-profiler
https://github.com/foundry-rs/starknet-foundry/blob/4a1aea0f9f888c9b4774af897fc010d029d0180c/crates/forge-runner/src/build_trace_data.rs#L41

Originated in: https://github.com/foundry-rs/starknet-foundry/pull/2191#discussion_r1701720675

### PR DESCRIPTION
**Pull request by piotmag769**
Closes #2355

I was bored during a weekend, don't ask xD

## Introduced changes

- Stop saving sierra for no reason - it made sense [in the past](https://github.com/foundry-rs/starknet-foundry/pull/2065), now Scarb gives us sierra files with only relevant `VersionedProgram` out of the box.

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by ddoktorski — COMMENTED** [review]
Generally the change looks good, but it seems to break `coverage` as it currently relies on the Sierra path [to get the project root](https://github.com/software-mansion/cairo-coverage/blob/e227a64b3920ae215b104b3986bf83c6258c0f30/crates/cairo-coverage/src/main.rs#L88) so it would be necessary to address this first

**Comment by piotmag769** [comment]
@ksew1 is supposed to fix this

**Review by piotmag769 — COMMENTED** [review]
LGTM (my PR)

File: crates/forge/src/test_filter.rs [hunk 1]
```diff
@@ -163,6 +163,7 @@ mod tests {
     fn filtering_tests() {
         let mocked_tests = TestTargetWithResolvedConfig {
             sierra_program: program_for_testing(),
+            sierra_program_path: Default::default(),
```
Comment (ksew1) [inline crates/forge/src/test_filter.rs:166]: Nit: I think it would be clearer to use `Arc::default()` so the reader knows exactly what type it is. The same applies to other similar occurrences.  
If you want to commit to this approach, you can disallow the `clippy::default_trait_access` lint, and Clippy will automatically highlight these places for you. (Should probably do this in separate pr)

**Review by ksew1 — APPROVED** [review]

**Review by cptartur — APPROVED** [review]
👍 
image

But please address the comment about `Default`

**Comment by cptartur** [comment]
cc @ddoktorski

**Comment by ddoktorski** [comment]
> But please address the comment about Default

https://github.com/foundry-rs/starknet-foundry/issues/2775
